CodePush.js

```diff
@@ -209,22 +209,23 @@ async function tryReportStatus(statusReport, resumeListener) {
     }
 
     NativeCodePush.recordStatusReported(statusReport);
-    resumeListener && AppState.removeEventListener("change", resumeListener);
+    appStateListener && appStateListener.remove();
   } catch (e) {
     log(`Report status failed: ${JSON.stringify(statusReport)}`);
     NativeCodePush.saveStatusReportForRetry(statusReport);
+    let localAppStateListener = appStateListener;
     // Try again when the app resumes
     if (!resumeListener) {
       resumeListener = async (newState) => {
         if (newState !== "active") return;
         const refreshedStatusReport = await NativeCodePush.getNewStatusReport();
         if (refreshedStatusReport) {
-          tryReportStatus(refreshedStatusReport, resumeListener);
+          tryReportStatus(refreshedStatusReport, resumeListener, localAppStateListener);
         } else {
-          AppState.removeEventListener("change", resumeListener);
+          localAppStateListener && localAppStateListener.remove();
         }
       };
-      AppState.addEventListener("change", resumeListener);
+      localAppStateListener = AppState.addEventListener("change", resumeListener);
```

Comment:
Not sure I understand this line. Basically token (aka localAppStateListener) from adding listener, as later on passed to scope of resumeListener right to remove it?

If so Im not sure I understand what line 216 is doing.

Reply:
Hello, thank you for your answer. What I wanted to do:
- initialize `localAppStateListener` with the `appStateListener ` parameter l.216
- assign it the return of `addEventListener` l.228
- in the event listener, access `localAppStateListener` which will hold the return from `addEventListener`

I agree that initializing `localAppStateListener` might be pointless. Do you want me to make this change or did you have something else in mind?